github/Auth.py

```diff
@@ -0,0 +1,342 @@
+############################ Copyrights and license ############################
+#                                                                              #
+# Copyright 2023 Enrico Minack <github@enrico.minack.dev>                      #
+#                                                                              #
+# This file is part of PyGithub.                                               #
+# http://pygithub.readthedocs.io/                                              #
+#                                                                              #
+# PyGithub is free software: you can redistribute it and/or modify it under    #
+# the terms of the GNU Lesser General Public License as published by the Free  #
+# Software Foundation, either version 3 of the License, or (at your option)    #
+# any later version.                                                           #
+#                                                                              #
+# PyGithub is distributed in the hope that it will be useful, but WITHOUT ANY  #
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS    #
+# FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more #
+# details.                                                                     #
+#                                                                              #
+# You should have received a copy of the GNU Lesser General Public License     #
+# along with PyGithub. If not, see <http://www.gnu.org/licenses/>.             #
+#                                                                              #
+################################################################################
+
+import abc
+import base64
+import datetime
+import time
+from datetime import timedelta
+from typing import Dict, Optional, Union
+
+import jwt
+
+from github import Consts
+from github.InstallationAuthorization import InstallationAuthorization
+from github.Requester import Requester, WithRequester
+
+# For App authentication, time remaining before token expiration to request a new one
+ACCESS_TOKEN_REFRESH_THRESHOLD_SECONDS = 20
+TOKEN_REFRESH_THRESHOLD_TIMEDELTA = timedelta(
+    seconds=ACCESS_TOKEN_REFRESH_THRESHOLD_SECONDS
+)
+
+
+class Auth:
```

Comment:
I think to be correct, this needs to also implement `ABC`
```suggestion
class Auth(abc.ABC):
```

Reply:
done